examples/connection_check.py

```diff
@@ -0,0 +1,59 @@
+"""
+How to check that connection credentials are suitable for queries and writes from/into specified bucket.
+"""
+
+from influxdb_client import InfluxDBClient
+from influxdb_client.client.write_api import SYNCHRONOUS
+from influxdb_client.rest import ApiException
+
+"""
+Define credentials
+"""
+url = "http://localhost:8086"
+token = "my-token"
+org = "my-org"
+bucket = "my-bucket"
+
+
+def check_connection():
+    """Check that the InfluxDB is running."""
+    print("> Checking connection ...", end=" ")
+    client.api_client.call_api('/ping', 'GET')
+    print("ok")
+
+
+def check_query():
+    """Check that the credentials has permission to query from the Bucket"""
+    print("> Checking credentials for query ...", end=" ")
+    try:
+        client.query_api().query(f"from(bucket:\"{bucket}\") |> range(start: -1m) |> limit(n:1)", org)
+    except ApiException as e:
+        # missing credentials
+        if e.status == 404:
+            raise Exception(f"The specified token doesn't have sufficient credentials to read from '{bucket}' "
+                            f"or specified bucket doesn't exists.") from e
+        raise
+    print("ok")
+
+
+def check_write():
+    """Check that the credentials has permission to write into the Bucket"""
+    print("> Checking credentials for write ...", end=" ")
+    try:
+        client.write_api(write_options=SYNCHRONOUS).write(bucket, org, b"")
```

Comment:
Is this tested on both OSS and Cloud2 ? I can imagine that the can be different order of 400 and 404 checks.

Reply:
I tested the latest version of InfluxDB v2.3.0+SNAPSHOT.090f681737 and InfluxDB Cloud 2. There is little inconsistency between query check that always fail with 404 (could not find bucket) but the write can fail with 403 or 404.

## **403** - bucket exists but you don't have a write permission
Reason: Forbidden
HTTP response body: {"code":"forbidden","message":"insufficient permissions for write"}

## **404** - bucket does not exists
Reason: Not Found
HTTP response body: {"code":"not found","message":"bucket \"xxx\" not found"}